Bug: all buttons in the sample of Kotlin&v3 cause a crash

#### Environment details

1. Specify the API at the beginning of the title (for example, "Places: ...")
v3
2. OS type and version
Emulator API 30
3. Library version and other environment information
    // GMS
    gmsImplementation 'com.google.android.libraries.places:places:2.4.0'

    // V3
    v3Implementation name:'places-maps-sdk-3.1.0-beta', ext:'aar'
    v3Implementation 'com.android.volley:volley:1.1.1'
    v3Implementation 'com.google.android.gms:play-services-base:17.4.0'
    v3Implementation 'com.google.android.gms:play-services-basement:17.4.0'
    v3Implementation 'com.google.android.gms:play-services-gcm:17.0.0'
    v3Implementation 'com.google.android.gms:play-services-location:17.1.0'
    v3Implementation 'com.google.android.gms:play-services-tasks:17.2.0'
    v3Implementation 'com.google.android.libraries.maps:maps:3.1.0-beta'
    v3Implementation 'com.google.auto.value:auto-value-annotations:1.7.4'
    v3Implementation 'com.google.code.gson:gson:2.8.6'
#### Steps to reproduce

  1. Import attached, which is based on the demo here:

https://github.com/googlemaps/android-places-demos/tree/main/demo-kotlin

#### Code example
It's just what's on the sample.

#### Stack trace
```
2020-10-19 10:43:01.502 32526-32526/com.example.placesdemo.v3 E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.example.placesdemo.v3, PID: 32526
    java.lang.NoClassDefFoundError: Failed resolution of: Lcom/google/android/datatransport/runtime/TransportRuntime;
        at com.google.android.libraries.places.internal.fg.<init>(PG:5)
        at com.google.android.libraries.places.internal.cb.h(PG:17)
        at com.google.android.libraries.places.internal.cb.i(PG:18)
        at com.google.android.libraries.places.internal.cb.l(PG:21)
        at com.google.android.libraries.places.internal.cb.r(PG:27)
        at com.google.android.libraries.places.internal.cb.a(PG:1)
        at com.google.android.libraries.places.api.Places.a(PG:43)
        at com.google.android.libraries.places.api.Places.createClient(PG:30)
        at com.example.placesdemo.CurrentPlaceTestActivity.onCreate(CurrentPlaceTestActivity.kt:62)
        at android.app.Activity.performCreate(Activity.java:8000)
        at android.app.Activity.performCreate(Activity.java:7984)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1309)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3422)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3601)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2066)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:223)
        at android.app.ActivityThread.main(ActivityThread.java:7656)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
     Caused by: java.lang.ClassNotFoundException: Didn't find class "com.google.android.datatransport.runtime.TransportRuntime" on path: DexPathList[[zip file "/data/app/~~4e9kwdXxgvYn4L0pRpwLiA==/com.example.placesdemo.v3-SLNdZABmDyGjIHSjoIMU4w==/base.apk"],nativeLibraryDirectories=[/data/app/~~4e9kwdXxgvYn4L0pRpwLiA==/com.example.placesdemo.v3-SLNdZABmDyGjIHSjoIMU4w==/lib/x86, /data/app/~~4e9kwdXxgvYn4L0pRpwLiA==/com.example.placesdemo.v3-SLNdZABmDyGjIHSjoIMU4w==/base.apk!/lib/x86, /system/lib, /system_ext/lib]]
```

Following these steps will guarantee the quickest resolution possible.

I've tried each of the buttons. All cause a crash.

See attached:

https://github.com/kataklisma/android-google-places/files/5399937/2020-10-19_10-42-11.zip

And the project (without the real key):

https://github.com/kataklisma/android-google-places/files/5399948/demo-kotlin.zip

--- a/demo-java/app/src/main/java/com/example/placesdemo/CurrentPlaceActivity.java
+++ b/demo-java/app/src/main/java/com/example/placesdemo/CurrentPlaceActivity.java
@@ -1,5 +1,5 @@
 /*
- * Copyright 2022 Google LLC
+ * Copyright 2023 Google LLC
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -26,19 +26,20 @@ import com.google.android.libraries.places.api.net.FindCurrentPlaceResponse;
 import com.google.android.libraries.places.api.net.PlacesClient;
 
 import android.Manifest.permission;
+import android.annotation.SuppressLint;
 import android.content.pm.PackageManager;
 import android.os.Bundle;
+import android.util.Log;
 import android.view.View;
-import android.widget.CheckBox;
-import android.widget.Toast;
 
 import java.util.List;
 
+import androidx.activity.result.ActivityResultLauncher;
+import androidx.activity.result.contract.ActivityResultContracts;
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 import androidx.annotation.RequiresPermission;
 import androidx.appcompat.app.AppCompatActivity;
-import androidx.core.app.ActivityCompat;
 import androidx.core.content.ContextCompat;
 
 import static android.Manifest.permission.ACCESS_FINE_LOCATION;
@@ -49,10 +50,41 @@ import static android.Manifest.permission.ACCESS_WIFI_STATE;
  */
 public class CurrentPlaceActivity extends AppCompatActivity {
 
+    private static final String TAG = "CURRENT_PLACE";
+
     private PlacesClient placesClient;
     private FieldSelector fieldSelector;
 
     private CurrentPlaceActivityBinding binding;
+
+    // [START maps_solutions_android_permission_request]
+    // Register the permissions callback, which handles the user's response to the
+    // system permissions dialog. Save the return value, an instance of
+    // ActivityResultLauncher, as an instance variable.
+    @SuppressLint("MissingPermission")
+    private final ActivityResultLauncher<String[]> requestPermissionLauncher =
+            registerForActivityResult(new ActivityResultContracts.RequestMultiplePermissions(), isGranted -> {
+                if (Boolean.TRUE.equals(isGranted.get(permission.ACCESS_FINE_LOCATION))
+                        && Boolean.TRUE.equals(isGranted.get(ACCESS_WIFI_STATE))) {
+                    findCurrentPlaceWithPermissions();
+                } else {
+                    // Fallback behavior if user denies permission
+                    Log.d(TAG, "User denied permission");
+                }
+            });
+    // [END maps_solutions_android_permission_request]
+
+    // [START maps_solutions_android_location_permissions]
+    @SuppressLint("MissingPermission")
+    private void checkLocationPermissions() {
+        if (ContextCompat.checkSelfPermission(this, ACCESS_FINE_LOCATION)
+                == PackageManager.PERMISSION_GRANTED) {
+            findCurrentPlaceWithPermissions();
+        } else {
+            requestPermissionLauncher.launch(new String[]{permission.ACCESS_FINE_LOCATION, permission.ACCESS_WIFI_STATE});
+        }
+    }
+    // [END maps_solutions_android_location_permissions]
 
     @Override
     protected void onCreate(@Nullable Bundle savedInstanceState) {
@@ -83,39 +115,13 @@ public class CurrentPlaceActivity extends AppCompatActivity {
         setLoading(false);
 
         // Set listeners for programmatic Find Current Place
-        binding.findCurrentPlaceButton.setOnClickListener((view) -> findCurrentPlace());
+        binding.findCurrentPlaceButton.setOnClickListener((view) -> checkLocationPermissions());
     }
 
     @Override
     protected void onSaveInstanceState(@NonNull Bundle bundle) {
         super.onSaveInstanceState(bundle);
         fieldSelector.onSaveInstanceState(bundle);
-    }
-
-    /**
-     * Fetches a list of {@link PlaceLikelihood} instances that represent the Places the user is
-     * most
-     * likely to be at currently.
-     */
-    private void findCurrentPlace() {
-        if (ContextCompat.checkSelfPermission(this, permission.ACCESS_WIFI_STATE)
-                != PackageManager.PERMISSION_GRANTED
-                || ContextCompat.checkSelfPermission(this, permission.ACCESS_FINE_LOCATION)
-                != PackageManager.PERMISSION_GRANTED) {
-            Toast.makeText(
-                            this,
-                            "Both ACCESS_WIFI_STATE & ACCESS_FINE_LOCATION permissions are required",
-                            Toast.LENGTH_SHORT)
-                    .show();
-        }
-
-        // Note that it is not possible to request a normal (non-dangerous) permission from
-        // ActivityCompat.requestPermissions(), which is why the checkPermission() only checks if
-        // ACCESS_FINE_LOCATION is granted. It is still possible to check whether a normal permission
-        // is granted or not using ContextCompat.checkSelfPermission().
-        if (checkPermission(ACCESS_FINE_LOCATION)) {
-            findCurrentPlaceWithPermissions();
-        }
     }
 
     /**
@@ -155,15 +161,6 @@ public class CurrentPlaceActivity extends AppCompatActivity {
         } else {
             return fieldSelector.getAllFields();
         }
-    }
-
-    private boolean checkPermission(String permission) {
-        boolean hasPermission =
-                ContextCompat.checkSelfPermission(this, permission) == PackageManager.PERMISSION_GRANTED;
-        if (!hasPermission) {
-            ActivityCompat.requestPermissions(this, new String[]{permission}, 0);
-        }
-        return hasPermission;
     }
 
     private boolean isDisplayRawResultsChecked() {

--- a/demo-kotlin/app/src/main/java/com/example/placesdemo/CurrentPlaceActivity.kt
+++ b/demo-kotlin/app/src/main/java/com/example/placesdemo/CurrentPlaceActivity.kt
@@ -1,5 +1,5 @@
 /*
- * Copyright 2022 Google LLC
+ * Copyright 2023 Google LLC
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -16,13 +16,14 @@
 package com.example.placesdemo
 
 import android.Manifest.permission
+import android.annotation.SuppressLint
 import android.content.pm.PackageManager
 import android.os.Bundle
 import android.view.View
 import android.widget.Toast
+import androidx.activity.result.contract.ActivityResultContracts
 import androidx.annotation.RequiresPermission
 import androidx.appcompat.app.AppCompatActivity
-import androidx.core.app.ActivityCompat
 import androidx.core.content.ContextCompat
 import com.example.placesdemo.databinding.CurrentPlaceActivityBinding
 import com.google.android.libraries.places.api.Places
@@ -39,6 +40,30 @@ class CurrentPlaceActivity : AppCompatActivity() {
     private lateinit var fieldSelector: FieldSelector
 
     private lateinit var binding: CurrentPlaceActivityBinding
+
+    @SuppressLint("MissingPermission")
+    val requestPermissionLauncher =
+        registerForActivityResult(
+            ActivityResultContracts.RequestMultiplePermissions()
+        ) { permissions ->
+            when {
+                permissions[permission.ACCESS_FINE_LOCATION] == true -> {
+                    // Precise location access granted.
+                    findCurrentPlaceWithPermissions()
+                }
+                permissions[permission.ACCESS_COARSE_LOCATION] == true -> {
+                    // Only approximate location access granted.
+                    findCurrentPlaceWithPermissions()
+                }
+                else -> {
+                    Toast.makeText(
+                        this,
+                        "Both ACCESS_COARSE_LOCATION & ACCESS_FINE_LOCATION permissions are required",
+                        Toast.LENGTH_SHORT
+                    ).show()
+                }
+            }
+        }
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
@@ -82,37 +107,22 @@ class CurrentPlaceActivity : AppCompatActivity() {
     }
 
     /**
-     * Fetches a list of [com.google.android.libraries.places.api.model.PlaceLikelihood] instances that represent the Places the user is
-     * most
-     * likely to be at currently.
+     * Check whether permissions have been granted or not, and ultimately proceeds to either
+     * request them or runs {@link #findCurrentPlaceWithPermissions() findCurrentPlaceWithPermissions}
      */
+    @SuppressLint("MissingPermission")
     private fun findCurrentPlace() {
-        if (ContextCompat.checkSelfPermission(this, permission.ACCESS_WIFI_STATE)
-            != PackageManager.PERMISSION_GRANTED
-            || ContextCompat.checkSelfPermission(this, permission.ACCESS_FINE_LOCATION)
-            != PackageManager.PERMISSION_GRANTED
-        ) {
-            Toast.makeText(
-                this,
-                "Both ACCESS_WIFI_STATE & ACCESS_FINE_LOCATION permissions are required",
-                Toast.LENGTH_SHORT
-            )
-                .show()
-        }
-
-        // Note that it is not possible to request a normal (non-dangerous) permission from
-        // ActivityCompat.requestPermissions(), which is why the checkPermission() only checks if
-        // ACCESS_FINE_LOCATION is granted. It is still possible to check whether a normal permission
-        // is granted or not using ContextCompat.checkSelfPermission().
-        if (ContextCompat.checkSelfPermission(
-                this,
-                permission.ACCESS_FINE_LOCATION
-            ) == PackageManager.PERMISSION_GRANTED
-        ) {
-            ActivityCompat.requestPermissions(this, arrayOf(permission.ACCESS_FINE_LOCATION), 0)
+        if (hasOnePermissionGranted(permission.ACCESS_FINE_LOCATION, permission.ACCESS_COARSE_LOCATION)) {
+            findCurrentPlaceWithPermissions()
             return
+        } else {
+            requestPermissionLauncher.launch(
+                arrayOf(
+                    permission.ACCESS_FINE_LOCATION,
+                    permission.ACCESS_COARSE_LOCATION
+                )
+            )
         }
-        findCurrentPlaceWithPermissions()
     }
 
     /**
@@ -152,4 +162,12 @@ class CurrentPlaceActivity : AppCompatActivity() {
     private fun setLoading(loading: Boolean) {
         binding.loading.visibility = if (loading) View.VISIBLE else View.INVISIBLE
     }
+
+    private fun hasOnePermissionGranted(vararg permissions: String): Boolean =
+        permissions.any {
+            ContextCompat.checkSelfPermission(
+                this,
+                it
+            ) == PackageManager.PERMISSION_GRANTED
+        }
 }

--- a/demo-kotlin/app/src/main/java/com/example/placesdemo/CurrentPlaceActivity.kt
+++ b/demo-kotlin/app/src/main/java/com/example/placesdemo/CurrentPlaceActivity.kt
@@ -47,18 +47,14 @@ class CurrentPlaceActivity : AppCompatActivity() {
             ActivityResultContracts.RequestMultiplePermissions()
         ) { permissions ->
             when {
-                permissions[permission.ACCESS_FINE_LOCATION] == true -> {
-                    // Precise location access granted.
-                    findCurrentPlaceWithPermissions()
-                }
-                permissions[permission.ACCESS_COARSE_LOCATION] == true -> {
+                permissions[permission.ACCESS_FINE_LOCATION] == true || permissions[permission.ACCESS_COARSE_LOCATION] == true -> {
                     // Only approximate location access granted.
                     findCurrentPlaceWithPermissions()
                 }
                 else -> {
                     Toast.makeText(
                         this,
-                        "Both ACCESS_COARSE_LOCATION & ACCESS_FINE_LOCATION permissions are required",
+                        "Either ACCESS_COARSE_LOCATION or ACCESS_FINE_LOCATION permissions are required",
                         Toast.LENGTH_SHORT
                     ).show()
                 }

